lightning/src/ln/channel.rs

```diff
@@ -3127,6 +3130,11 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		self.our_htlc_minimum_msat
 	}
 
+	/// Allowed in any state (including after shutdown)
+	pub fn get_our_htlc_maximum_msat(&self) -> Option<u64> {
+		self.our_htlc_maximum_msat
```

Comment:
I think we should always return an htlc_maximum_msat by looking at the channel value (and never returning more than, say, 90% of it whether it was set in an option or not).

Reply:
I implemented this, but I'm not sure if I understood you 100%. Please double-check.